internal/maintenance/startstop/start_stop.go

```diff
@@ -5,6 +5,23 @@ import (
 	"sync"
 )
 
+// Service is a generalized interface for a service that starts and stops,
+// usually one backed by embedding BaseStartStop.
+type Service interface {
```

Comment:
Move in from the maintenance package so it can be reused. Required by `startstoptest.Stress`, but will also likely be useful more broadly.